packages/next/client/components/redirect.ts

```diff
@@ -0,0 +1,14 @@
+export const REDIRECT_ERROR_CODE = 'NEXT_REDIRECT'
+
+class Redirect extends Error {
```

Comment:
This probably shouldn't extend error because that causes it to track a stack frame which can be somewhat expensive.

It would also get caught by `instanceof Error` checks where as we might prefer it to be rethrown.